docs/deployments.md

```diff
@@ -170,3 +164,39 @@ Additionally, the following environment variables are provided by OpenShift to t
 * `OPENSHIFT_DEPLOYMENT_NAMESPACE` - the namespace of the `replicationController` representing the new `deployment`
 
 The replica count of the `replicationController` for the new deployment will be 0 initially. The responsibility of the `strategy` is to make the new `deployment` live using whatever logic best serves the needs of the user.
+
+## Rollbacks
+
+Rolling a deployment back to a previous state is a two step process accomplished by:
+
+1. POSTing a `rollback` API object to a special endpoint, which generates and returns a new `deploymentConfig` representing the rollback state
+2. POSTing the new `deploymentConfig` to the API server
+
+The `rollback` API object configures the generation process and provides the scope of the rollback. For example, given a previous deployment `deployment-1` and the current deployment `deployment-2`:
+
+```
+{
+  "metadata": {
+    "name": "rollback-1"
+  },
+  "kind": "DeploymentConfigRollback",
+  "apiVersion": "v1beta1",
+  "spec": {
+    "deploymentRef": {
+      "name": "deployment-1",
+      "namespace": "default"
```

Comment:
See my pull, I have lots of validations on them - I think there are downsides to validating too strongly, but I went ahead and added those now and we can relax them later.

----- Original Message -----
> > +1. POSTing a `rollback` API object to a special endpoint, which generates
> > and returns a new `deploymentConfig` representing the rollback state
> > +2. POSTing the new `deploymentConfig` to the API server
> > +
> > +The `rollback` API object configures the generation process and provides
> > the scope of the rollback. For example, given a previous deployment
> > `deployment-1` and the current deployment `deployment-2`:
> > +
> > +```
> > +{
> > +  "metadata": {
> > +    "name": "rollback-1"
> > +  },
> > +  "kind": "DeploymentConfigRollback",
> > +  "apiVersion": "v1beta1",
> > +  "spec": {
> > +    "deploymentRef": {
> > +      "name": "deployment-1",
> > +      "namespace": "default"
> 
> I like `from`. Also, right now I'm anchoring all lookups to the
> `deploymentRef` namespace- did we nail down what the rules there should be
> (e.g. always use request namespace and ignore ref namespace, prefer ref
> namespace if present, etc.)?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/648/files#r23386576